src/spock_recovery.c

```diff
@@ -477,6 +510,137 @@ drop_recovery_slot(void)
 	PG_END_TRY();
 }
 
+/*
+ * advance_recovery_slot_to_min_position
+ *
+ * Advance the recovery slot's confirmed_flush_lsn to the minimum position
+ * across all active peer subscriptions. This ensures the slot stays behind
+ * the slowest subscriber, allowing historical replay for rescue operations.
+ *
+ * This function should be called periodically by the manager worker.
+ */
+void
+advance_recovery_slot_to_min_position(void)
+{
+	SpockRecoverySlotData *slot_data;
+	ReplicationSlot *recovery_slot;
+	char slot_name[NAMEDATALEN];
+	XLogRecPtr min_remote_lsn = InvalidXLogRecPtr;
+	XLogRecPtr current_slot_lsn;
+	List *subscriptions;
+	ListCell *lc;
+
+	if (!SpockRecoveryCtx)
+		return;
+
+	/* Get recovery slot info from shared memory */
+	LWLockAcquire(SpockRecoveryCtx->lock, LW_SHARED);
+	slot_data = get_recovery_slot();
+	if (!slot_data->active)
+	{
+		LWLockRelease(SpockRecoveryCtx->lock);
+		return;
+	}
+	strlcpy(slot_name, slot_data->slot_name, NAMEDATALEN);
+	LWLockRelease(SpockRecoveryCtx->lock);
+
+	/* Start a transaction to query subscriptions */
+	StartTransactionCommand();
+
+	/* Get all active non-rescue subscriptions */
+	subscriptions = get_node_subscriptions(InvalidOid, false);
+
+	/*
+	 * Find the minimum remote_lsn across all peer subscriptions.
+	 * This represents the slowest subscriber's position.
+	 */
+	foreach(lc, subscriptions)
+	{
+		SpockSubscription *sub = (SpockSubscription *) lfirst(lc);
+		Oid origin_id;
+		XLogRecPtr remote_lsn;
+		char origin_name[NAMEDATALEN];
+
+		/* Skip rescue subscriptions */
+		if (sub->rescue_temporary)
+			continue;
+
+		/* Skip disabled subscriptions */
+		if (!sub->enabled)
+			continue;
+
+		/* Build origin name for this subscription */
+		snprintf(origin_name, NAMEDATALEN, "spk_pgedge_%s_sub_%s_%s",
+				 sub->origin->name, sub->origin->name, sub->target->name);
+
+		/* Look up replication origin */
+		origin_id = replorigin_by_name(origin_name, true);
+		if (origin_id == InvalidRepOriginId)
+			continue;
+
+		/* Get remote LSN from origin status */
+		replorigin_session_setup(origin_id, 0);
+		replorigin_session_reset();
+		remote_lsn = replorigin_session_get_progress(false);
+
```

Comment:
Is it possible that session_setup/reset here can messup with apply manager's session? Maybe getting the lsn from `pg_replication_origin_status` or progress table can be more safe?